lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp

```diff
@@ -758,7 +759,23 @@ Fvp::PrimSelections MayaHydraSceneIndex::UfePathToPrimSelections(const Ufe::Path
     // Not the best implementation performance-wise, as ufeToDagPath converts
     // the UFE path to a string, then does a Dag path lookup with the string.
     constexpr bool isSprim = false; // Can't handle sprims as of 15-Aug-2023.
-    SdfPath primPath = GetPrimPath(UfeExtensions::ufeToDagPath(appPath), isSprim);
+    auto dagPath = UfeExtensions::ufeToDagPath(appPath);
+
+    SdfPath primPath = GetPrimPath(dagPath, isSprim);
+    
+    //Check if this maya node has a special SdfPath associated with it, this is for custom or maya usd data producers scene indices.
+    //The class MhDataProducersMayaNodeToSdfPathRegistry does a mapping between Maya nodes and USD paths.
+    //The maya nodes registered in this class are used by data producers as a parent to all
+    //primitives. This class is used when the user selects one of these
+    //maya nodes to return the matching SdfPath so that all prims child of this maya node are
+    //highlighted.
+    MDagPath shapeDagPath(dagPath);
+    shapeDagPath.extendToShape();
+    const SdfPath matchingPath = MhDataProducersMayaNodeToSdfPathRegistry::Get().GetPath(MObjectHandle(shapeDagPath.node()));
+    if (! matchingPath.IsEmpty()) {
+        primPath = matchingPath;
+    }
```

Comment:
Just thinking out loud, but I feel like it's a bit counterintuitive to have the `MayaHydraSceneIndex`, which AFAIK is for Maya data, handle the node selection for all other data producers. Though in a way since they're Maya nodes it also makes sense? I don't really have a clear solution in mind either, maybe we could move some stuff into the Maya implementation layer of the data producer interface? Maybe it'd be easier to address this separately though.